capstone-rs/src/instruction.rs

```diff
@@ -36,6 +36,19 @@ pub type RegIdInt = u16;
 #[derive(Copy, Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
 pub struct RegId(pub RegIdInt);
 
+/// Represents how the register is accessed.
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq)]
+pub enum RegAccessType {
```

Comment:
Can an access be not read nor write, but still valid?

I also wonder if we should eliminate the `Invalid` variant and return `Option<RegAccessType>`, which may make it less likely for users to ignore the "invalid" case.